docs/sources/mimir/migration-guide/migrating-from-cortex.md

```diff
@@ -228,76 +232,32 @@ You can update to the Grafana Mimir Helm chart from the Cortex Helm chart.
-   - Set `ruler.alertmanager_url` to `'dnssrvnoa+http://_http-metrics._tcp.{{ template "mimir.fullname" . }}-alertmanager-headless.{{ .Release.Namespace }}.svc.cluster.local/alertmanager'`.
-   - If you want to use memberlist as the ring KV store, set `memberlist.join_members` to `['{{ include "mimir.fullname" . }}-gossip-ring']`.
-   - Append the caching configuration to `blocks_storage.bucket_store`.
-   - Set `ingester.ring.num_tokens` to the existing value you set in `ingester.lifecycler.num_tokens`.
-     This is especially important if you are using the default Cortex Helm chart values as it sets this to `512`.
+   > **Note:** We added `mimir.structuredConfig` in mimir-distributed v3.0.0. This new field allows overriding specific
+   > configuration without the need to rewrite the whole block string literal like in `mimir.config`.
 
-   A partial Helm values file with the changes incorporated looks similar to:
+   In your Helm values file:
 
    ```yaml
    mimir:
-     config: |
-       blocks_storage:
-         {{- if index .Values "memcached-chunks" "enabled" }}
-         chunks_cache:
-           backend: "memcached"
-           memcached:
-             addresses: dns+{{ template "mimir.fullname" . }}-memcached-chunks.{{ .Release.Namespace }}.svc:{{ (index .Values "memcached-chunks").port }}
-             max_item_size: {{ mul (index .Values "memcached-chunks").maxItemMemory 1024 1024 }}
-         {{- end }}
-         {{- if index .Values "memcached-metadata" "enabled" }}
-         metadata_cache:
-           backend: "memcached"
-           memcached:
-             addresses: dns+{{ template "mimir.fullname" . }}-memcached-metadata.{{ .Release.Namespace }}.svc:{{ (index .Values "memcached-metadata").port }}
-             max_item_size: {{ mul (index .Values "memcached-metadata").maxItemMemory 1024 1024 }}
-         {{- end }}
-         {{- if index .Values "memcached-queries" "enabled" }}
-         index_cache:
-           backend: "memcached"
-           memcached:
-             addresses: dns+{{ template "mimir.fullname" . }}-memcached-index-queries.{{ .Release.Namespace }}.svc:{{ (index .Values "memcached-index-queries").port }}
-             max_item_size: {{ mul (index .Values "memcached-index-queries").maxItemMemory 1024 1024 }}
-         {{- end }}
-       frontend_worker:
-         frontend_address: "{{ template "mimir.fullname" . }}-query-frontend.{{ .Release.Namespace }}.svc:{{ include "mimir.serverGrpcListenPort" . }}"
-       ingester:
-         ring:
-           num_tokens: 512
-       memberlist:
-         join_members: ["{{ include "mimir.fullname" . }}-gossip-ring"]
-       ruler:
-         alertmanager_url: "dnssrvnoa+http://_http-metrics._tcp.{{ template "mimir.fullname" . }}-alertmanager-headless.{{ .Release.Namespace }}.svc.cluster.local/alertmanager"
+     structuredConfig: <CONFIGURATION FILE CONTENTS>
```

Comment:
hmm, I feel like we should give some example so people see how indent things and how it will look like

Reply:
Added an example. However the position `<CONFIGURATION FILE CONTENTS>` is always placed there by `make docs`.